File: google-code-jam/2011/round1b/A.java
import java.util.*;
import java.io.*;
public class A {
    public static PrintStream out = System.out;
    public static void main(String args[]) {
        Scanner sc = new Scanner(System.in);
        int T = sc.nextInt();
        for (int tc = 1; tc <= T; tc++) {
            int N = sc.nextInt();
            data = new char[N][];
            for (int i = 0; i < N; i++) {
                data[i] = sc.next().toCharArray();
            }
            String ans = "";
            for (int i = 0; i < N; i++) {
                // System.out.println(WP(i) + " " + OWP(i) + " " + OOWP(i));
                ans += "\n" + (0.25 * WP(i) + 0.50 * OWP(i) + 0.25 * OOWP(i));
            }
            out.printf("Case #%d: %s\n", tc, ans);
        }
    }
    public static char[][] data;
    public static double WP(int i, int j) {
        int tot = 0;
        int win = 0;
        for (int k = 0; k < data.length; k++) {
            if (k == j) continue;
            if (data[i][k] != '.') tot++;
            if (data[i][k] == '1') win++;
        }
        // System.out.println("i: " + i + " j: " + j + " " + win + "/" + tot);
        return ((double) win) / tot;
    }
    public static double WP(int i) {
        return WP(i, -1);
    }
    public static double OWP(int i) {
        double totWP = 0.0;
        int tot = 0;
        for (int j = 0; j < data.length; j++) {
            if (data[i][j] != '.') {
                tot++;
                totWP += WP(j, i);
            }
        }
        return totWP / tot;
    }
    public static double OOWP(int i) {
        double totOWP = 0.0;
        int tot = 0;
        for (int j = 0; j < data.length; j++) {
            if (data[i][j] != '.') {
                tot++;
                totOWP += OWP(j);
            }
        }
        return totOWP / tot;
    }
}

File: google-code-jam/2011/round1b/A.py
'''
Problem
@CONTEST Google Code Jam 2011: Round 1B
@ID A
@NAME RPI
@URL http://code.google.com/

Solution
@AUTHOR Wonjohn Choi (http://capcs.wordpress.com/)
@LANG Python 3
@RESULT accepted (small input, large input)
'''

def solve():
    fout = open('C:\\Users\\Wonjohn Choi\\Python\\Google Code Jam 2011 - Round 1B\\A.out','w')
    fin = open('C:\\Users\\Wonjohn Choi\\Python\\Google Code Jam 2011 - Round 1B\\A.in','r')

    t = int(next(fin).strip())
    for i in range(t):
        n = int(next(fin).strip())
        data = []
        for j in range(n):
            data.append(next(fin).strip())
            
            #print(data[j])
        WP = []
        OWP = []
        OOWP = []
        
        #WP
        for j in range(n):
            win = 0
            lose = 0
            for k in range(n):
                if data[j][k]=='.':
                    pass
                elif data[j][k]=='1':
                    win+=1
                else:
                    lose+=1
            WP.append(win/(win+lose))
        
        #OWP
        for j in range(n):
            owp = 0
            nn = 0
            for k in range(n):
                if k!=j and data[k][j]!='.':
                    win = 0
                    lose = 0
                    for l in range(n):
                        if l!=j:
                            if data[k][l]=='.':
                                pass
                            elif data[k][l]=='1':
                                win+=1
                            else:
                                lose+=1
                    owp+=win/(win+lose)
                    nn+=1
            OWP.append(owp/nn)
        
        #OOWP
        for j in range(n):
            oowp = 0
            nn = 0
            for k in range(n):
                if k!=j and data[k][j]!='.':
                    oowp+=OWP[k]
                    nn+=1
            OOWP.append(oowp/nn)
        
        fout.write('Case #%d:\n'%(i+1))
        #final
        for j in range(n):
            fout.write(str(0.25 * WP[j] + 0.50 * OWP[j] + 0.25 * OOWP[j]))
            fout.write('\n')
                            
solve()